streaming/src/main/scala/org/apache/spark/streaming/receiver/ReceiverSupervisorImpl.scala

```diff
@@ -136,14 +136,18 @@ private[streaming] class ReceiverSupervisorImpl(
       metadataOption: Option[Any],
       blockIdOption: Option[StreamBlockId]
     ) {
+    var rBlock = receivedBlock
     val blockId = blockIdOption.getOrElse(nextBlockId)
     val numRecords = receivedBlock match {
       case ArrayBufferBlock(arrayBuffer) => arrayBuffer.size
+      case IteratorBlock(iterator) => 
+        var arrayBuffer =  ArrayBuffer(iterator.toArray : _*)		
```

Comment:
Agree, but Custom Receiver storing blocks using iterator can not utilize the UI stats . Not sure if there is any wide usage of Iterator , but in my Kafka Consumer (http://spark-packages.org/package/dibbhatt/kafka-spark-consumer) I earlier used  Iterator but now changed it to ArrayBuffer. But people who use Iterator will have this issue..